lib/packages_file.dart

```diff
@@ -142,12 +145,24 @@ void write(StringSink output, Map<String, Uri> packageMapping,
     if (baseUri != null) {
       uri = _relativize(uri, baseUri);
     }
-    output.write(uri);
     if (!uri.path.endsWith('/')) {
-      output.write('/');
```

Comment:
This was a problem causing `package:file://path/to/folder#dart=2.4/`

Reply:
Good catch.